compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1253,6 +1253,11 @@ rustc_queries! {
         feedable
     }
 
+    query struct_target_features(def_id: DefId) -> &'tcx Vec<Symbol> {
```

Comment:
Makes more sense to use `&'tcx [Symbol]` since the reference means the `Vec` is immutable so we gain nothing from Vec's ability to change length

Reply:
The main reason for using `Vec` here is that it is clone-able and hence cache-able - AFAIU &[] doesn't play well with arena_cache, and also AFAIU I cannot not use it as otherwise the returned data doesn't have an owner... I think

Reply:
You can allocate the slice on an arena in `TyCtxt` via `tcx.arena.alloc_slice_from_iter`, it's a pretty common pattern in rustc to return `&'tcx [T]` from queries.